toolchain/check/testdata/impl/no_prelude/import_interface_assoc_const.carbon

```diff
@@ -0,0 +1,1104 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/impl/no_prelude/import_interface_assoc_const.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/impl/no_prelude/import_interface_assoc_const.carbon
+
+// --- interface.carbon
+library "[[@TEST_NAME]]";
+
+interface I { let T:! type; }
+
+interface I3 {
+  let T1:! type;
+  let T2:! type;
+  let T3:! type;
+}
+
+interface NonType {
+  let Y:! {.a: {}};
+}
+
+// --- basic.carbon
+library "[[@TEST_NAME]]";
+import library "interface";
+
+impl () as I where .T = {} { }
```

Comment:
This test will stop working once we implement the orphan rule. Maybe define a class in this file and implement the interface for it instead? (Likewise below.)

Reply:
Done.